libcore/src/test/java/com/mapbox/android/core/metrics/AbstractCompositeMetricsTest.java

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.android.core.metrics;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AbstractCompositeMetricsTest {
+  private TestCompositeMetrics metrics;
+
+  @Before
+  public void setUp() {
+    metrics = new TestCompositeMetrics(TimeUnit.HOURS.toMillis(1));
+  }
+
```

Comment:
Should add multi threads test cases.

Reply:
sure. i'm putting together basic test cases, we shall have an integration ticket where test coverage should be expanded, we will address them in the next PR